inference-engine/samples/benchmark_app/main.cpp

```diff
@@ -53,6 +53,9 @@ bool ParseAndCheckCommandLine(int argc, char* argv[]) {
         throw std::logic_error("Model is required but not set. Please set -m option.");
     }
 
+    if (FLAGS_b != 0 && !FLAGS_tensor_shape.empty()) {
+        throw std::logic_error("Batch size for dynamic models should be specified in tensor shape option.");
```

Comment:
If user passes both -b and -tensor_shape parameters how it should be resolved? 
For example:
-b 3 -tensor_shape [1,3,128,128][3,3,128,128]
Should we force batch size to 3 for each tensor shape and get two equal shapes?

Reply:
With -b and -shape parameters, batch size is taken from -b.

Reply:
@ivikhrev I don't see a problem with combining -b and --tensor_shape. For example, I have topology with [-1, 3, -1, -1] shape. And I would like to reshape it first to [batch_size, 3, -1, -1] and then variate spatial shapes using --tensor_shape parameter. Moreover, this scenario should work for --shape parameter already. As for your example, it just non valid combination of parameter values. I think. that batch should not be applied to tensor shape. It should be used as a shape for model reshaping only.

Reply:
I don't quite understand. Right now input blobs are constructed based on tensor shapes, including batch size. And you said that 

> I think, that batch should not be applied to tensor shape. It should be used as a shape for model reshaping only.

How then should input blobs be constructed?
I see only variant, when if `-b` and `-tensor_shape` were provided, we would force batch in tensor shapes to the valur of `-b`

Reply:
@ivikhrev 
1.  Previously we had no --tensor_shape parameter for blobs. But we had -b and -shape for model shapes. -b is just the simplest case modify batch only in shape. So from this point of view, I don't see why we should add special limitations for -b only.
-shape [batch_size,3,-1,-1]  is equal to -b batch_size when other dimensions should not be changed.
2. If you would like to apply -b parameter to --tensor_shape it is new functionality. It can be discussed. But currently, I'm talking about misalignment -b and -shape behavior only.
3. Mismatch between --shape data and --tensor_shape can happen not only in batch size. For example, --shape [-1, 3, 400, 400] --tensor_shape [1,3,128,128] So it is not batch related. And I think it is user repososibility provide correct data.

Reply:
Ok, I'll align behavior for `-b `and `-tensor_shape`. 
But one last clarification from your side:
if next arguments were passed: `-shape [-1, 3, -1, -1]  -b batch_size`, what correct tensor shape would be ?
only those where batch is equal to `batch_size` ( -tensor_shape [batch_size, 3, 128, 128] ), and what to do if it is not,, should we throw an exception in this case?

Reply:
To summarize: **old behavior:** -b replaces batch size in input shapes provided by -shape or by network info.
**New behavior:** -b replaces batch size in input shapes provided by -shape or by network info (for backward compatibility in static scenarios). In case of using -tensor_shape parameter and -b error will be shown, cause user has to provide full information in -tensor_shape parameter, otherwise it won't work. So, -b is extra here anyway and we should now allow using it with -tensor_shape parameter to avoid misunderstanding.
**The only case where we might use -b (or similar parameter) in case of dynamic shapes** is the case of multiple input images without providing -tensor_shape parameter (currently not implemented due to old api limitations). I.e. H,W, and C will be read from image, but batch for tensor shape has to be provided manually. But in this case I'd like to make separate parameter (something like -tensor_shape_batch) or hardcode batch 1 if tensor shape is not provided (this will make usage easier because several consequent images might have differnet shapes, so in case of batch>1 we'll have to detect those problems).

-

Reply:
Removed restriction on `-b` and `-tensor_shape `https://github.com/openvinotoolkit/openvino/pull/7788/commits/ab0e82e3367c8fb2dee2c6dfd4172b64944748be